test/gst-va/encode/12bit/hevc.py

```diff
@@ -0,0 +1,99 @@
+###
+### Copyright (C) 2023 Intel Corporation
+###
+### SPDX-License-Identifier: BSD-3-Clause
+###
+
+from .....lib import *
+from .....lib.gstreamer.va.util import *
+from .....lib.gstreamer.va.encoder import EncoderTest
+
+spec = load_test_spec("hevc", "encode", "12bit")
+
+@slash.requires(*have_gst_element("vah265dec"))
+class HEVC12EncoderBaseTest(EncoderTest):
+  def before(self):
+    super().before()
+    vars(self).update(
+      codec         = "hevc-12",
+      gstdecoder    = "vah265dec",
+      gstmediatype  = "video/x-h265",
+      gstparser     = "h265parse",
+    )
+
+  def get_file_ext(self):
+    return "h265"
+
+@slash.requires(*have_gst_element("vah265enc"))
+@slash.requires(*platform.have_caps("encode", "hevc_12"))
+class HEVC12EncoderTest(HEVC12EncoderBaseTest):
+  def before(self):
+    super().before()
+    vars(self).update(
+      caps      = platform.get_caps("encode", "hevc_12"),
+      gstencoder= "vah265enc",
+      lowpower  = False,
+    )
+
+class cqp(HEVC12EncoderTest):
+  def init(self, tspec, case, gop, slices, bframes, qp, quality, profile):
+    vars(self).update(tspec[case].copy())
+    vars(self).update(
+      bframes   = bframes,
+      case      = case,
+      gop       = gop,
+      profile   = profile,
+      qp        = qp,
+      quality   = quality,
+      rcmode    = "cqp",
+      slices    = slices,
+    )
+
+  @slash.parametrize(*gen_hevc_cqp_parameters(spec, ['main12']))
+  def test(self, case, gop, slices, bframes, qp, quality, profile):
+    self.init(spec, case, gop, slices, bframes, qp, quality, profile)
+    self.encode()
+
+class cbr(HEVC12EncoderTest):
+  def init(self, tspec, case, gop, slices, bframes, bitrate, fps, profile):
+    vars(self).update(tspec[case].copy())
+    vars(self).update(
+      bframes   = bframes,
+      bitrate   = bitrate,
+      case      = case,
+      fps       = fps,
+      gop       = gop,
+      maxrate   = bitrate,
+      minrate   = bitrate,
+      profile   = profile,
+      rcmode    = "cbr",
+      slices    = slices,
+    )
+
+  @slash.parametrize(*gen_hevc_cbr_parameters(spec, ['main12']))
+  def test(self, case, gop, slices, bframes, bitrate, fps, profile):
+    self.init(spec, case, gop, slices, bframes, bitrate, fps, profile)
+    self.encode()
+
+class vbr(HEVC12EncoderTest):
+  def init(self, tspec, case, gop, slices, bframes, bitrate, fps, quality, refs, profile):
+    vars(self).update(tspec[case].copy())
+    vars(self).update(
+      bframes   = bframes,
+      bitrate   = bitrate,
+      case      = case,
+      fps       = fps,
+      gop       = gop,
+      maxrate   = int(bitrate / 0.7),
+      minrate   = bitrate,
```

Comment:
minrate will be greater than maxrate, see another comment.

Reply:
see reply in https://github.com/intel/vaapi-fits/pull/491#discussion_r1160829998